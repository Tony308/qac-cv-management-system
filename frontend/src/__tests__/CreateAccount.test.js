import React from 'react';
import { render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import AccountContainer from '../containers/CreateAccountContainer';

afterEach(cleanup);

describe('Create account test', () => {
  test('submit form without function', () => {
    const {getByRole, getByPlaceholderText, getByText} = render(
      <AccountContainer/>
    );

    const username = getByRole('textbox');
    const password = document.getElementById('password');
    const confirm = getByPlaceholderText('Confirm Password');

    userEvent.type(username, 'username');
    userEvent.type(password, 'password');
    userEvent.type(confirm, 'pass');

    const message = getByText("Passwords don't match");

    expect(message).toHaveTextContent("Passwords don't match");
    userEvent.type(confirm, 'word');
    
    expect(confirm.value).toBe('password');
    expect(username.value).toBe('username');
    expect(password.value).toBe('password');    
  })

  test('error message prop', () => {
    const error = "Error Message";
    const {getByText} = render(
      <AccountContainer errMessage={error}/>
    );
    const message = getByText('Error Message');
    expect(message).toHaveTextContent('Error Message');
  })  
})

test('snapshot test', () => {
  const tree = renderer.create(<AccountContainer/>).toJSON();
  expect(tree).toMatchSnapshot();
})