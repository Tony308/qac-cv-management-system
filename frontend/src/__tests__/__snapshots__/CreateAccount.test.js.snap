// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div>
  <h1>
     Create Account
  </h1>
  <p
    style={
      Object {
        "color": "red",
        "fontSize": "14px",
      }
    }
  >
    <b />
  </p>
  <form
    method="POST"
    name="create-account-form"
  >
    <p>
      Username:
    </p>
    <input
      id="username"
      name="username"
      placeholder="Username"
      type="text"
    />
    <p>
      Password:
    </p>
    <input
      id="password"
      name="password"
      placeholder="Password"
      type="password"
    />
    <p
      style={
        Object {
          "color": "red",
          "fontSize": "12px",
        }
      }
    />
    <p>
      Confirm Password:
    </p>
    <input
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
    />
    <br />
    <br />
    <input
      type="submit"
      value="Create Account"
    />
  </form>
</div>
`;
